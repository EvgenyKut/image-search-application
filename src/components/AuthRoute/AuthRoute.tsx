import { ReactElement } from 'react';
import { Navigate } from 'react-router-dom';
import { useAppSelector } from '../../hooks';

const AuthRoute = ({ children }: { children: ReactElement }) => {
  const { isAuth } = useAppSelector((state) => state.authReducer);
  return isAuth ? children : <Navigate to="/" />;
};

export default AuthRoute;
