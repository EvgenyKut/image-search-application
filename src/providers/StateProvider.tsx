import React, { ReactElement } from 'react';
import { Provider } from 'react-redux';
import { setupStore } from '../store/store';

type StateProviderProps = {
  children: ReactElement;
};
const StateProvider: React.FC<StateProviderProps> = ({ children }) => {
  const store = setupStore();
  return <Provider store={store}>{children}</Provider>;
};

export default StateProvider;
