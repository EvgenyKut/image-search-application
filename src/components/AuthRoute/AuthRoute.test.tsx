import renderer from 'react-test-renderer';
import AuthRoute from './index';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import StateProvider from '../../providers/StateProvider';

describe('Container-component render test', () => {
  it('snapshot', () => {
    const tree = renderer
      .create(
        <StateProvider>
          <BrowserRouter>
            <Routes>
              <Route
                element={
                  <AuthRoute>
                    <></>
                  </AuthRoute>
                }
              />
            </Routes>
          </BrowserRouter>
        </StateProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
