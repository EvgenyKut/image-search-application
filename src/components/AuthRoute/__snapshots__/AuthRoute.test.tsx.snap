// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container-component render test snapshot 1`] = `null`;
